packages/dataviews/src/dataviews.tsx

```diff
@@ -21,30 +21,34 @@ import { VIEW_LAYOUTS } from './layouts';
 import BulkActions from './bulk-actions';
 import { normalizeFields } from './normalize-fields';
 import BulkActionsToolbar from './bulk-actions-toolbar';
-import type { Action, AnyItem, Field, View, ViewBaseProps } from './types';
+import type { Action, Field, View, ViewBaseProps } from './types';
 
-interface DataViewsProps< Item extends AnyItem > {
+type ItemWithId = { id: string };
+
+type DataViewsProps< Item > = {
```

Comment:
I'm not really familiar with any of this. Where is `Item` defined? Below in `Item extends ItemWithId` right? I thought an `Item` was an item in the data array? Those don't have a `getItemId` function?

Reply:
I guess I'm not getting this part

```ts
( Item extends ItemWithId
 	? { getItemId?: ( item: Item ) => string }
 	: { getItemId: ( item: Item ) => string } )
```

Reply:
`Item` is a type parameter, it's completely generic. `DataViews` can work with any data type whatsoever, you can have a dataview of `number`s. As long as you provide functions to extract IDs and field values from the item.

The `Item extends ItemWithId` part says that if the `Item` is an object that has an `id` field, the `getItemId` prop is optional, because we have a suitable default for it.

But if `Item` is not compatible with `ItemWithId`, then `getItemId` is mandatory. Because the default (`item => item.id`) is not going to work with such a type.